vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/FieldSteps.java

```diff
@@ -104,6 +120,46 @@ public WebElement clearFieldLocatedByUsingKeyboard(Locator locator)
         return element;
     }
 
+    /**
+     * Gets a list of elements by <b>locator</b> and checks that it is
+     * sorted by text with using <b>sortingOrder</b> by element text.
+     * List should have more than 1 elements. If some elements don't
+     * contain text it will be displayed in the report.
+     * @param locator locator to locate elements
+     * @param sortingOrder sorting order<br>
+     * <i>Available rule:</i>
+     * <ul>
+     * <li><b>ASCENDING</b> - will check that the list is sorted alphabetically,
+     * <li><b>DESCENDING</b> - will check that the list is sorted reverse alphabetically
+     * </ul>
+     */
+    @Then("elements located `$locator` are sorted by text in $sortingOrder order")
+    public void areElementSorted(Locator locator, SortingOrder sortingOrder)
+    {
+        List<WebElement> elements = baseValidations.assertIfAtLeastNumberOfElementsExist(
+                String.format("Fields by locator: %s", locator), locator, 2);
+        if (!elements.isEmpty())
+        {
+            List<String> textsOfElements = elements.stream()
+                    .map(webElementActions::getElementText)
+                    .collect(Collectors.toList());
+            if (softAssert.assertNotNull("The texts of elements was found", textsOfElements))
+            {
+                for (int i = 0; i < textsOfElements.size(); i++)
+                {
+                    softAssert.assertTrue(String.format("The field number %d contains non empty text", i + 1),
+                            StringUtils.isNotEmpty(textsOfElements.get(i)));
+                }
+                textsOfElements.removeAll(Arrays.asList(null, StringUtils.EMPTY));
```

Comment:
double iteration, you can remove items from the list while you traversing it in the for-loop